core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -122,6 +123,23 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
   // and applications between check task and clean task.
   private val pool = ThreadUtils.newDaemonSingleThreadScheduledExecutor("spark-history-task-%d")
 
+  private val diskStoreCacheEnabled = conf.get(LOCAL_STORE_DIR).nonEmpty &&
+    conf.get(DISK_BACKGROUND_CACHE_ENABLED)
+
+  // Create a threadpool for background disk caching, if enabled,
+  // "spark.history.fs.disk.backgroundCache.enabled"
+  private val replayPool = if (diskStoreCacheEnabled) {
```

Comment:
this usage of something.get is kinda odd...

Reply:
Thanks @felixcheung. Yeah. I would like to know if the approach is okay, code wise still need some clean up.